### PR TITLE
Added BigBirdPegasus onnx config

### PR DESCRIPTION
**Pull request by nandwalritik**
# What does this PR do?

Added BigBirdPegasus OnnxConfig to make this model available for conversion.
@ChainYo 

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
https://github.com/huggingface/transformers/issues/16308
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- ~~[ ] Did you write any new necessary tests?~~

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/bigbird_pegasus/configuration_bigbird_pegasus.py
Comment (ChainYo) [inline src/transformers/models/bigbird_pegasus/configuration_bigbird_pegasus.py:360]: I'm not sure about this. I think it should be added to `src/transformers/onnx/config.py` and not directly in the `BigBirdPegasus` config file.

**Comment by ChainYo** [comment]
Hello @nandwalritik nice PR! 

I'm tagging maintainers for discussion @lewtun @sgugger

File: src/transformers/models/bigbird_pegasus/configuration_bigbird_pegasus.py
Comment (nandwalritik) [reply src/transformers/models/bigbird_pegasus/configuration_bigbird_pegasus.py:360]: I saw a similar implementation for BART and mBART, in both the models it was in the configuration file of the model itself. Let me know if any changes are required, I will integrate them accordingly.

Comment (ChainYo) [reply src/transformers/models/bigbird_pegasus/configuration_bigbird_pegasus.py:360]: Ok I faced the question for `LayoutLMv2` ONNX Config implementation (I still need to finish this :roll_eyes: ) and the dummy inputs generation was made on the onnx file.
Anyways let's see what maintainers think about this! :+1:

Comment (lewtun) [reply src/transformers/models/bigbird_pegasus/configuration_bigbird_pegasus.py:360]: Yes @nandwalritik is correct - for BART we had to override the default functions to generate dummy data, so what is done here looks good to me

File: tests/onnx/test_onnx_v2.py
Comment (lewtun) [inline tests/onnx/test_onnx_v2.py:204]: Could you please confirm that the slow tests pass with 

```
RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -s -k "bigbird-pegasus"
```

**Review by lewtun — APPROVED** [review]
Thanks a lot for adding support for this model @nandwalritik! The implementation LGTM and I think we can merge once you've confirmed the slow tests pass and rebased on `main` to account for a recent refactoring to order all the model names in `features.py` alphabetically

Gently pinging @sgugger for final approval 🤗

File: src/transformers/onnx/features.py
Comment (lewtun) [inline src/transformers/onnx/features.py:177]: After rebasing, please put this in alphabetical order, i.e. just below `bigbird`

**Review by sgugger — APPROVED** [review]
LGTM with @lewtun 's comments. Thanks a lot for adding this!

File: src/transformers/onnx/features.py
Comment (nandwalritik) [reply src/transformers/onnx/features.py:168]: Rebased with main branch and modified this according to alphabetical order.

File: tests/onnx/test_onnx_v2.py
Comment (nandwalritik) [reply tests/onnx/test_onnx_v2.py:204]: On running above command none of the test cases are being selected. Are there no test cases for this model?

```(nlp) nandwalritik@hp:~/transformers$ RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -s -k "bigbird-pegasus" -W ignore
2022-05-04 23:13:23.976033: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2022-05-04 23:13:23.976055: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
WARNING:absl:No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
============================================================================ test session starts ============================================================================
platform linux -- Python 3.9.5, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/nandwalritik/transformers, configfile: setup.cfg
plugins: hypothesis-6.46.1, timeout-2.1.0, dash-2.3.1, forked-1.4.0, xdist-2.5.0
collecting ... 2022-05-04 23:13:25.956129: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:936] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2022-05-04 23:13:25.956484: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2022-05-04 23:13:25.956555: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcublas.so.11'; dlerror: libcublas.so.11: cannot open shared object file: No such file or directory
2022-05-04 23:13:25.956637: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcublasLt.so.11'; dlerror: libcublasLt.so.11: cannot open shared object file: No such file or directory
2022-05-04 23:13:25.956718: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcufft.so.10'; dlerror: libcufft.so.10: cannot open shared object file: No such file or directory
2022-05-04 23:13:25.956798: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcurand.so.10'; dlerror: libcurand.so.10: cannot open shared object file: No such file or directory
2022-05-04 23:13:25.956877: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcusolver.so.11'; dlerror: libcusolver.so.11: cannot open shared object file: No such file or directory
2022-05-04 23:13:25.956939: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcusparse.so.11'; dlerror: libcusparse.so.11: cannot open shared object file: No such file or directory
2022-05-04 23:13:25.957002: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudnn.so.8'; dlerror: libcudnn.so.8: cannot open shared object file: No such file or directory
2022-05-04 23:13:25.957012: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1850] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2022-05-04 23:13:25.957210: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
collected 194 items / 194 deselected / 0 selected                                                                                                                           

========================================================================== 194 deselected in 0.11s ==========================================================================

```

Comment (lewtun) [reply tests/onnx/test_onnx_v2.py:204]: Oh indeed, it seems that the `-k` argument of `pytest` only detects tests with an underscore. Can you please try running:

```
RUN_SLOW=1 pytest tests/onnx/test_onnx_v2.py -s -k "bigbird_pegasus"
```

File: src/transformers/onnx/features.py
Comment (lewtun) [reply src/transformers/onnx/features.py:168]: Thank you! It seems there are still a few merge conflicts to resolve though and the PR has picked up many changes from the rebase. Would you mind resolving them please?

Comment (nandwalritik) [reply src/transformers/onnx/features.py:168]: Okay I will resolve these merge conflicts.

File: tests/onnx/test_onnx_v2.py
Comment (nandwalritik) [reply tests/onnx/test_onnx_v2.py:204]: All the 8 test cases passed successfully after running this command.

**Pull request closed without merging**
